### ISSUE TITLE
InsertHtml is unable to render ordered list if it's a neighbor of unordered list

### ISSUE THREAD

**Issue by monominia** [OfficeDev/office-js#3967]
> office-js bot:
Internal issue 8860502 has been created.

# Provide required information needed to triage your issue

## Your Environment

* Platform [PC desktop, Mac, iOS, Office on the web]: Office on the web
* Host [Excel, Word, PowerPoint, etc.]: Word
* Office version number: N/A
* Operating System: Both Mac OS and Windows
* Browser (if using Office on the web): Latest Chrome
 
## Expected behavior

When calling `insertHtml` with argument like `<ul><li>Milk</li></ul><ol><li>Grace</li></ol>`, I'm expecting a result like [this in MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/ul).

## Current behavior

when calling insertHtml with string like `<ul><li>one</li></ul><ol><li>two</li></ol>` the rendered result will be an unordered list as a whole.

## Steps to reproduce

1. Download this public gist [link](https://gist.github.com/monominia/92dbf5553e5305a1d53450c335bdd28b)
2. Import it inside Script Lab
3. Click the `Insert the content control` button.

## Link to live example(s)

1. Live example in Script Lab: [link](https://seismic-my.sharepoint.com/:w:/p/jqi/EQ3olTtSPY9BlnM3W9kD5PUBM1_MKZPNB4gPsZVJLwn8Aw?e=gQRc1n)

# Provide additional details

1. ______
2. ______
3. ______

## Context

We're building a Word add-in, which allowed the users to edit inside the Content Control field and then use the add-in to save what they typed in as a rich text field inside our service. (50,000 MAU)

## Useful logs

- [ ] Console errors
- [x] Screenshots
![Screenshot 2023-12-18 101940](https://github.com/OfficeDev/office-js/assets/13508893/c407304f-7a83-4066-bdf3-cf360a11c5c1)
- [ ] Test file (if only happens on a particular file)

Thank you for taking the time to report an issue. Our triage team will respond to you in less than 72 hours. Normally, response time is <10 hours Monday through Friday. We do not triage on weekends.
This issue was closed Duplicate

**Comment by monominia** [comment]
Hey @yilin4, could you take a look at this for us?

**Comment by zhenhuangMSFT** [comment]
Hi @monominia , your issue is well received. We are aware that it is a follow-up of another issue you posted before. We are still investigating into it. Please allow some time before we can get back to you. Thanks!

**Comment by xiruatms** [comment]
@monominia , correct me if I have misunderstanding here:

![image](https://github.com/OfficeDev/office-js/assets/40648203/640f804c-a057-4d5a-b4a5-cd27a9981017)

**Comment by monominia** [comment]
Hey @xiruatms , that ul page has a bug, try https://developer.mozilla.org/en-US/docs/Web/HTML/Element/ol
![Screenshot 2024-01-18 at 10 20 40 AM](https://github.com/OfficeDev/office-js/assets/13508893/94b63853-7c61-42f7-ac1f-73b7fb4a5de6)

**Comment by JayS-Novisto** [comment]
Hello! We are experimenting the same issue in our Add-In. Any update on that ticket?
![IssueBulletPoint](https://github.com/OfficeDev/office-js/assets/131690490/468a5d89-18ff-40ee-890d-5884605d6e18)
![IssueBulletPoint-Expected](https://github.com/OfficeDev/office-js/assets/131690490/f6cb28ed-19f6-42dc-baba-be714a73aa31)

**Comment by monominia** [comment]
Hey Microsoft team, could you help us take a look? Thanks!

**Comment by zhenhuangMSFT** [comment]
Hi @monominia, we are contacting owner team internally. Will update here if anything. Thanks for your patience!

**Comment by monominia** [comment]
Thanks @zhenhuangMSFT !

**Comment by monominia** [comment]
Hey @gergzk , is this fixed?

**Comment by zhenhuangMSFT** [comment]
Hi @monominia, this has not been fixed. We are still tracking this issue with partner team with internal Id 8805895.

**Comment by jaiswalshubh1** [comment]
@monominia - an honest correction, and an apology. You were told on this thread that this was still being tracked with the partner team. It was subsequently closed internally without a fix, and nobody came back here to tell you.

Where it actually stands: this was root-caused. The problem is in the logic that decides how pasted list content is merged, and two distinct breaking scenarios were identified and understood. A specific check in that path is what produces the behaviour, and the engineer who traced it recorded that they could not establish why that check was there in the first place. It was closed without that question being answered.

That is not a satisfactory basis for closing a reproduced defect in a documented API, particularly one that another developer on this thread also reported hitting.

cc: @kustonaut
